### DISCUSSION TITLE
FOCEI crashes on Apple Silicon Mac

### DISCUSSION THREAD

**Discussion by TingjieGuo** [nlmixr2/nlmixr2#75] [General]
Hi all,

I'm using a M1 Mac (MacOS 11.7.1) on which an R 4.2.1 intel version (i.e. using Apple's Rosetta) is installed. I also have a x86 gfortran 10.2 compiler configured and the nlmixr2 was installed from source (compiled locally, factory clang). But R crashes all the time when using focei method at the step of calculating $\partial (f)/\partial (\eta)$ with R prompting "R encountered a fatal error". It works all fine with nlme and saem method. It seems to me a focei-specific issue. Does anyone have an idea on this? I used the example code provided here (https://nlmixr2.org/reference/covarSearchAuto.html?q=cov#null). Many thanks!

Best,
Tingjie

**Comment by mattfidler** [comment]
Hi @TingjieGuo 

A crash in this specific location often occurs because of `symengine`.  I would suggest compiling that package from source or using the newly released `symengine` binaries.  They were having some issues with the M1 mac, so they recently released a new package.

Let me know how it goes.

**Reply by TingjieGuo** [reply]
Hi @mattfidler, I got a bit trouble compiling symengine locally. It complains about the missing MPFR library while I do have it installed. I have posted a question there (https://github.com/symengine/symengine.R/issues/118). Do you perhaps happen to know something about it?

**Comment by mattfidler** [comment]
I believe you can install from homebrew:

Once you have installed homebrew on your mac you use:

```sh
brew   install cmake gmp mpfr libmpc
```

From the iTerm command line

**Reply by mattfidler** [reply]
I read the comment and I am unsure.  @john-harrold has a m1 mac and maybe he has some tips

**Comment by TingjieGuo** [comment]
Here's an update. This was an issue caused by the mismatch of architecture between libraries and packages involved. Simply put, one should configure everything including R, gfortran, symengine and its dependencies, and importantly homebrew(!) built for the same architecture either x86 or arm64. Homebrew was what I got unrightfully that caused all issues. I installed a x86 build of R and gfortran and a universal Symengine. Symengine requires cmake gmp mpfr libmpc and homebrew is recommended for installing these libraries. However, I installed the Homebrew in the default way that recognized my M1 Mac and hence anything I installed using it was of arm64 build. I then reinstalled R and gfortran of arm64 build and expectedly Symengine was successfully compiled locally (though compilation is not necessarily needed I think). And FOCE finally worked. I didn't test if it would work having everything of x86 build on my M1 Mac but I suspect it will. For that I would have to also have x86 build of Homebrew to make sure everything I install using it will be of x86 build. There's a bunch of tutorials of how one can install Homebrew x86 build on a M1 Mac out there on the internet.

**Reply by mattfidler** [reply]
Thanks for documenting this @TingjieGuo 

It will be helpful for others too.
